packages/@css-blocks/glimmer/src/ElementAnalyzer.ts

```diff
@@ -133,19 +137,43 @@ export class ElementAnalyzer {
     }
   }
 
-  *eachAnalyzedAttribute(node: AnalyzableNode): Iterable<[string, string, AST.AttrNode | AST.HashPair]> {
+  private _throwIfInvalidAttribute(node: AnalyzableNode, property: AnalyzableProperty, name: string): void {
+    if (isPathExpression(property) && name === "class") {
+      throw cssBlockError(`The namespace:class attribute must contain a value and is forbidden to be positional. Did you mean block:class="foo"?`, node, this.template);
+    }
+  }
+
+  private _getAnalyzableAttributeName(attribute: AnalyzableProperty): string {
+    return isAttrNode(attribute) ? attribute.name :
+            isHashPair(attribute) ? attribute.key :
+              isPathExpression(attribute) ? attribute.original :
+                "";
```

Comment:
This is a case where `assertNever()` is ideal. Because the type guards are exhaustive, the empty string case should never happen. And if we did add more types to the `AnalyzableProperty` alias, we'd want the compiler to tell us about this code so we can add custom logic for it.

```suggestion
    if (isAttrNode(attribute)) {
      return attribute.name;
    } else if (isHashPair(attribute)) {
      return attribute.key;
    } else if (isPathExpression(attribute)) {
      return attribute.original;
    } else {
      assertNever(attribute);
    }
```

Reply:
It's not actually exhaustive. The "" will be for any attribute that isn't one of those. Like anything on a html element for example. Or any non-supported attribute.

I actually use this in the other PR for noting the unsupported attributes.

Reply:
Updated!